.github/workflows/continuous-integration.yml

```diff
@@ -29,12 +25,26 @@ jobs:
       run: python -m pip install --upgrade pip -r test-requirements.txt
     - name: Run pytest
       run: pytest
+  check-style:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@master
     - name: Check style
-      run: flake8
+      uses: py-actions/flake8@v2
+  check-code-formatting:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@master
     - name: Check code formatting
-      run: black --check --diff .
+      uses: psf/black@stable
+  code-cov:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@master
     - name: Upload coverage to Codecov
-      if: matrix.python-version == 3.10 && matrix.os == 'ubuntu-latest'
```

Comment:
I'd like to keep this `if` so coverage is not uploaded for each job in the matrix (only one is needed).

Reply:
Ah, now I get it, it's only performed once :)